File: Adventofcode/day1.py
#!/usr/bin/env python
import adventapi

x=adventapi.levelInput(1)
a=0
p2=False
for k,v in enumerate(x):
    if v=='(': a+=1
    elif v==')': a-=1
    if a==-1 and not p2: 
        print "part 2:",k+1
        p2=True
    
print "part 1:", a
    

File: Adventofcode/2018/day1.rb
#!/usr/bin/env ruby
require_relative 'adventapi'

input = AdventApi::get_day(1,2018).lines.map(&:to_i)

# part 1
puts input.inject(:+)

# part 2 - refactor maybe later? but it does work.
x = 0
seen = [0]
while seen.uniq == seen
	input.each do |i|
		x += i
		if seen.include? x
			seen << x
			break
		end
		
		seen << x
	end
end

puts x